src/coreclr/vm/jitinterface.cpp

```diff
@@ -1545,9 +1548,41 @@ void CEEInfo::getFieldInfo (CORINFO_RESOLVED_TOKEN * pResolvedToken,
             {
                 fieldAccessor = CORINFO_FIELD_STATIC_ADDRESS;
 
+                // Allocate space for the local class if necessary, but don't trigger
+                // class construction.
+                DomainLocalModule* pLocalModule = pFieldMT->GetDomainLocalModule();
+                pLocalModule->PopulateClass(pFieldMT);
+
                 // We are not going through a helper. The constructor has to be triggered explicitly.
-                if (!pFieldMT->IsClassPreInited())
+                if (!pFieldMT->IsClassInited())
                     fieldFlags |= CORINFO_FLG_FIELD_INITCLASS;
+
+                GCX_COOP();
+
+                pResult->fieldLookup.addr = pField->GetStaticAddressHandle((void*)pField->GetBase());
+                if (fieldFlags & CORINFO_FLG_FIELD_STATIC_IN_HEAP)
+                {
+                    Object* frozenObj = VolatileLoad((Object**)pResult->fieldLookup.addr);
+
+                    if (frozenObj == nullptr)
+                    {
+                        // Boxed static is not yet set, allocate it
+                        pFieldMT->AllocateRegularStaticBox(pField, (BYTE*)pResult->fieldLookup.addr);
```

Comment:
Do we have any tests that exercise this path?

Reply:
I didn't put any tests for it because even this simple sample hits it:
```csharp

using System.Runtime.CompilerServices;

class Proga
{
    static DateTime dt = DateTime.Now;

    [MethodImpl(MethodImplOptions.NoInlining)]
    static DateTime GetField() => dt;

    static void Main()
    {
        for (int i = 0; i < 60; i++)
        {
            Thread.Sleep(16);
            GetField();
        }
    }
}
```
AvaloniaILSpy app hits this path 50 times during start.

Reply:
Ok, I just wanted make sure that it is getting hit.